.pre-commit-config.yaml

```diff
@@ -34,3 +37,9 @@ repos:
       # Only if we want to create a requirements.txt file
       # - id: poetry-export
       #   args: ["-f", "requirements.txt", "-o", "requirements.txt"]
+
+  - repo: https://github.com/econchick/interrogate
+    rev: 1.5.0
+    hooks:
+      - id: interrogate
+        args: [--fail-under=70, --verbose, --generate-badge=./docs/_static]
```

Comment:
70 is the % of functions that need doc strings for this to pass. `--generate-badge` creates an svg in the folder that we then point to in the read me. Its possible that this will be annoying as it essnetailly mean committing twice sometimes (you will see what I mean when you try committing)